spec/support/match_xml.rb

```diff
@@ -0,0 +1,36 @@
+def match_xml(expected, ignore_list)
+  MatchXml.new(expected, ignore_list)
+end
+
+class MatchXml
+  def initialize(expected, ignore_list)
+    @ignore_list = ignore_list
+    @expected = MatchXml.xml_to_css_path_hash(expected, @ignore_list)
+  end
+
+  def matches?(actual)
+    @actual = MatchXml.xml_to_css_path_hash(actual, @ignore_list)
+    RSpec::Matchers::BuiltIn::Include.new(@expected).matches?(@actual)
+  end
+
+  def failure_message
+    RSpec::Support::Differ.new.diff(@actual, @expected)
+  end
+
+  class << self
+    def traverse_elements(base_node, ignore_list, &block)
```

Comment:
see if this can just be a wrapper around `Node#traverse` that applies the ignore list

Reply:
so close, but it can't: for some reason the way #traverse yields nodes means that their css_paths come out as long strings of "nth-of-type(x)" and such

Reply:
this was wrong - it was the namespaces that were messing with the paths, for some reason. nonetheless, i'm still using my own recursion because i want to avoid recursing into ignored non-leaf nodes, and `#traverse` doesn't let me do that.